src/ln/msgs.rs

```diff
@@ -435,6 +439,9 @@ pub trait ChannelMessageHandler : events::EventsProvider + Send + Sync {
 	/// understand or indicate they require unknown feature bits), no_connection_possible is set
 	/// and any outstanding channels should be failed.
 	fn peer_disconnected(&self, their_node_id: &PublicKey, no_connection_possible: bool);
+
+	//Testing
+	fn push_event(&self, event: events::Event);
```

Comment:
Please at least put a #[cfg(test)] here so that there isn't a testing function in a pub trait.